purchases/src/main/kotlin/com/revenuecat/purchases/common/offerings/OfferingsFactory.kt

```diff
@@ -42,7 +41,18 @@ internal class OfferingsFactory(
                     productIds = allRequestedProductIdentifiers,
                     onCompleted = { productsById ->
                         try {
-                            logMissingProducts(allRequestedProductIdentifiers, productsById)
+                            val notFoundProductIds = allRequestedProductIdentifiers
+                                .filterNot { productsById.containsKey(it) }
+                                .toSet()
+                            notFoundProductIds
+                                .takeIf { it.isNotEmpty() }
+                                ?.let { missingProducts ->
```

Comment:
Hmm should this be? 
```suggestion
                                ?.also { missingProducts ->
```

I'm wondering if `notFoundProductIds` will have a non-null value with the `let` (maybe it's working not sure)

Reply:
Ah right, sorry I misread it, this should be fine yeah!